### DISCUSSION TITLE
После прошивки ESP32-S3 не реагирует на управление

### DISCUSSION THREAD

**Discussion by Naglietroll** [vvip-68/LedPanelWiFi#64] [Вопросы и ответы]
Доброго.

Благодарность автору проекта за труды.
Захотел попробовать ESP32-S3, много памяти всё такое. Версия с 16 Мб Flash, 8 Мб PSRAM. Думал всё будет как в первой версии прошивки, но нет.

IDE 1.8.19
Версия ядра ESP32 2.0.14
Версия FastLED 3.6.0
Настройки и библиотеки при прошивке в соответствии с мануалом.

Файлов систему выбирал 8M spiffs (3MB APP/ 1.5MB SPIFFS), т.к. на 4MB with spiff(1.2MB APP/1.5MB SPIFFS) писало что мало места, а этот вариант единственный со SPIFFS, на 16MB будет уже FATFS.
По пинам ничего не менял, да и базово в прошивке тоже, кроме параметров матрицы (32х8). Периферия вся включена, но при запуске физические не подключена. При запуске подключал только матрицу к плате на G14 (собственно как указано в WEB UI). После прошивки WEB UI поднимается, но сама матрица светиться всеми светодиодами одновременно в рандомном порядке, не в соответствии с режимом выбранным в WEB UI и в принципе на команды смену режимов из WEB UI никак не реагирует.

ЧЯДНТ? Помогите разобраться куда копать, что делать.

**Comment by vvip-68** [comment]
О! Вы вступили на путь неизведанного. Стандартный ESP32 - это совсем не то же самое, что ESP32 S3
Тут и пины по другому могут использоваться и свои нюансы могут быть...
В общем  - если кто собирал именно на  S3 - может подскажут, ждите.
Или лучше сами попробуйте разобраться и сказать что пошло не так.

Скорее всего выбран неподходящий пин для вывода... Попробуйте другой пин.
НУ и также не исключен вариант некорректного питания системы - то есть неподходящее напряжение, несогласованность уровней (цифровой шум), шумящий источник питания и все проблемы, связанные с питанием.

**Reply by Naglietroll** [reply]
Понял. Спасибо за советы. Буду еще экспериментировать, если будут результаты отпишусь. Ну или возможно сообщество подскажет.
Тогда вопрос в догонку: а нет ли проверенных и опробованных вариантов, что бы не стандартная ESP32, но и не неизведанная ESP32 S3? Что-то где памяти хватает и может заработать без танцев с бубном? Или же все что кроме стандартной ESP 32 это шаг в неизведанное?

**Reply by vvip-68** [reply]
Фактически - да. Я экспериментировал с некоторыми платами. Не помню была ли среди них S3? но у меня точно все работало с теми, что проверял.  Кроме стандартной S32 я остановился вот на такой (ESP32 Mini D1)
![image](https://github.com/user-attachments/assets/0564a8cd-d5f5-4868-89c0-cfaa9a6f39db)
или такой (ESP32 S2 mini)
![image](https://github.com/user-attachments/assets/68215c9c-3a28-42e0-ba0a-229969504212)

Плата ESP32 S3 вроде у меня валялась где-то и даже, кажется с ней все работало (но это не точно).
Сейчас на ней нет собранного стенда, а экспериментировать пока нет времени.
Другие дела.

**Reply by Naglietroll** [reply]
> Фактически - да. Я экспериментировал с некоторыми платами. Не помню была ли среди них S3? но у меня точно все работало с теми, что проверял. Кроме стандартной S32 я остановился вот на такой (ESP32 Mini D1) ![image](https://private-user-images.githubusercontent.com/39353139/352018336-0564a8cd-d5f5-4868-89c0-cfaa9a6f39db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxNjMwOTUsIm5iZiI6MTcyMjE2Mjc5NSwicGF0aCI6Ii8zOTM1MzEzOS8zNTIwMTgzMzYtMDU2NGE4Y2QtZDVmNS00ODY4LTg5YzAtY2ZhYTlhNmYzOWRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI4VDEwMzMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiMmViNDY5NDhkNzE2OGQ4NTk5N2JmMDkzNWMwMGEzYjM3ZDY1NzEzNzI3ODU3MjI2MmRhOGY2M2YxNTM3ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GWMfbmreTqwDch-q1FrlkbiF2F6blqRhDeczuXcLfNE) или такой (ESP32 S2 mini) ![image](https://private-user-images.githubusercontent.com/39353139/352018399-68215c9c-3a28-42e0-ba0a-229969504212.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxNjMwOTUsIm5iZiI6MTcyMjE2Mjc5NSwicGF0aCI6Ii8zOTM1MzEzOS8zNTIwMTgzOTktNjgyMTVjOWMtM2EyOC00MmUwLWJhMGEtMjI5OTY5NTA0MjEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI4VDEwMzMxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzMjVhNzIxYjhlMDAwMWFlZjU4ZjAzNzE4ZTcyMmFjNWQyZDE1NzI3YzZmODBmMzJjN2M0YjA1OTIyY2YzYTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.t8-EbfsPI1ydV1mKu81eSnB5x7Bk4lcGa3gvLw21cVk)
> 
> Плата ESP32 S3 вроде у меня валялась где-то и даже, кажется с ней все работало (но это не точно). Сейчас на ней нет собранного стенда, а экспериментировать пока нет времени. Другие дела.

Спасибо. Это по сути те же 4МБ что и на стандартной ESP32/ Если не получиться S3 домучать (тоже время только на выходных есть), то буду Mini делать. Если тема не будет закрыта и что-то получиться решить с S3 - отпишусь.

**Reply by vvip-68** [reply]
Ну, в общем, я не знаю что и как вы там делаете.
Я не поленился, достал эту плату из загашника, подключил. Настройки (и файловая система - как у Вас).

![image](https://github.com/user-attachments/assets/fa2f8746-6b27-482d-b9b1-b0d1bccaecce)


Все оборудование цеплять не стал, хотя оно кроме MP3-плеера активировано в настройках, подключил только вывод на матрицу.
Всё прекрасно работает на S3 - завелось с первого раза.

https://drive.google.com/file/d/1Ag0k6Poky6etOzE5o7GBbhuFk-O3LnGR/view?usp=sharing

**Reply by vvip-68** [reply]
И да - чип на плате у меня M0N8R2

**Reply by Naglietroll** [reply]
Доброго времени.

Спасибо.
Сел пока выходной еще раз попробовать, раз пришла новая платка. У меня она кстати несколько по другому выглядит чем по ссылке (видимо надо брать фиолетовую, потому как черные, как у меня даже на алике дешевле). Зашел в тему посмотреть, а тут оказывается есть еще ответ. Извиняюсь, не видел.
Перепроверял проект перед прошивкой и возник вопрос: самом скетче версия ядра для ESP32 указана 2.0.14, а в мануале 1.0.6. Так какая все же верная? И если 2.0.14, то на ней тоже нужно выполнять: 
    Перейдите в папку C:\Users\<user>\AppData\Local\Arduino15\packages\esp32\hardware\esp32\1.0.6\ и далее найдите файлы:
    \cores\esp32\main.cpp, \tools\sdk\sdkconfig, \tools\sdk\include\config\sdkconfig.h
    В каждом из этих файлов найдите строчку - параметр CONFIG_ARDUINO_LOOP_STACK_SIZE и измените значение по умолчанию 8192 на 16138, пересоберите проект с новым значением параметра?
Аналогично версия FastLED. В мануале 3.5.0, в скетче 3.4.0-3.6.0, с приоритетом последней версии. Какой вариант верный.?Возможно помните при какой комбинации удалось достичь удачного результата?

**Reply by vvip-68** [reply]
Конечно уже не помню, столько времени прошло.
Актуальная версия ядра на которой работало 2.0.14, соответственно то что прописано для ядра 1.0.6 делать вроде бы не нужно.
Там же была описана ситуация, которую эти исправления лечат? Ну, соответственно, если будет возникать такая ситуация - значит поменять. Но, кажется, уже не нужно всё же...
FastLed - 3.6 - в последнее время в проект были добавлены какие-то изменения, которых в 3.4 и 3.5 еще нет - не скомпилится.
Может быть когда-нибудь придет время, я вернусь к проекту, изучу / вспомню что где прописано и внесу изменения...
Но это не точно...

**Reply by Naglietroll** [reply]
Доброго времени.
В общем дело было в питании. Т.к. пробовалось запускаться в тестовом режиме, то питания на МК подавалось через USB, а на матрицу с блока питания, и вот в таком случае матрица выводила просто рандомные цвета. А вот если все питать уже с макетки одним блоком питания, всё работает. При этом у вас на видео вы матрицу питаете с МК, а на него подаете питание через USB. Так вот у меня так не взлетало.
Всю периферию проверил. Не инициализируется только SD карта. От сюда вопрос: возможно подскажете проверенный shield? Т.к. не думаю что мог подключить как-то не так. Да и проверил несколько раз. И еще возможно подскажите наиболее легкий путь наделать пользовательских анимаций под матрицу 32*8 из имеющихся в теме исходников? А то сделанных уже анимаций много, но все не под тот размер. Или же если не возможно, то вообще наиболее короткий путь сделать под свой размер.
Спасибо.

**Reply by vvip-68** [reply]
1. Симптомы очень похожи, когда не соединены минусы питания контроллера и матрицы.
2. Где у меня матрица питается через Микроконтроллер от USB? Это в принципе невозможно. Если питание на матрицу идет через USB микроконтроллера - ток потребления на матрице 16x16 примерно 15 ампер, что много превышает предельный ток защитного диода USB - 500 ма и он немедленно сгорает с синим дымком. После чего по USB уже контроллер не распознается, не прошивается и не видится.
3. В проекте есть утилита JinxFramer - там можно покадрово рисовать анимации и далее сохранять их либо в формате SD-карты (файлы Jinx *.out) либо в виде кода, который встраивается в сам код скетча как например Марио. Но память контроллера очень ограничена и таких (в виде кода) много анимаций не вставишь...
4. Я на всех своих устройствах с поддержкой SD-карты использовал такой шилд:  https://aliexpress.ru/item/32718621622.html
![image](https://github.com/user-attachments/assets/2c8be905-06fc-4dca-88b6-2746a777cc9c)
Он отлично встает этажеркой на Wemos d1 mini, Wemos d1 mini 32 (https://aliexpress.ru/item/1005006237136185.html) или ESP32 S2 Mini (https://aliexpress.ru/item/1005007671098191.html)

**Reply by Naglietroll** [reply]
1. Так и было
2. Не матрица. У вас на видео выше микроконтроллер от USB, а матрица с микроконтроллера берет питание. У меня так не взлетало. Взлетело когда оба с одного блока питания.
3. Будем пробовать. Вот из-за ограниченной памяти МК и хотел подключить SD.
4. Вот именно такой и не взлетел. Не знаю почему. Пины дважды проверил. Питание с МК 3.3 + GND.

**Reply by vvip-68** [reply]
2. На видео выше просто не видно, что с блока питания идет толстый провод на матрицу (оно под матрицей) и с точностью до наоборот - не матрица берет питание с контроллера, а контроллер берет питание с матрицы.
4. А вы попробуйте еще +5 к этому шилду подпаять (не убирая +3.3)

**Reply by Naglietroll** [reply]
2. Теперь всё стало на свои места. 
3. Чуда не случилось. Всё решилось заменой SD карты на самсунговскую. С ней заработало просто от 3.3V/
Ещё раз спасибо за помощь и терпение.
